### PR TITLE
https://issues.redhat.com/browse/ACM-13718

### PR DESCRIPTION
**Pull request by dockerymick**

### CODE REVIEW

File: observability/observability_alerts.adoc [hunk 1]
```diff
@@ -55,6 +60,86 @@ route:
...
+
+. To set an expiration of 365 days for your `cert.crt` and `cert.key` files, run the following command:
+
++
+[source,bash]
+----
+openssl x509 -req -days 365 -in cert.csr -signkey cert.key -out cert.crt
+----
+
+//I added the steps to create the .crt and .key files because I saw that in Polarian. Should cert.csr be changed to cert.crt in the openssl req command?
```
Comment (quchangl-github) [inline observability/observability_alerts.adoc]: @dockerymick I added setps regrading how to gerenrate cert.crt and cert.key in Polarion for testing, want to provide more detailed steps for the tester. I think it's fine not to add these two commands of creating key and cert files.

File: observability/observability_alerts.adoc [hunk 2]
```diff
@@ -55,6 +60,86 @@ route:
...
++
+[source,yaml]
+----
+...
+advancedConfig: 
+ alertmanager: 
+   secrets: ['tls']
+----
+
+//Should "['tls']" in the previous sample be moved to a new line? Asking because the sample appears that way in Polarian
```
Comment (quchangl-github) [inline observability/observability_alerts.adoc]: the format is fine, but `advancedConfig` should be changed to `advanced`

File: observability/observability_alerts.adoc [hunk 3]
```diff
@@ -55,6 +60,86 @@ route:
...
+----
+"global":
+  "resolve_timeout": "5m"
+  "http_config":
+    "tls_config":
+      "cert_file": "/etc/alertmanager/secrets/storyverify/tls.crt"
+      "key_file": "/etc/alertmanager/secrets/storyverify/tls.key"
+----
+
+//also got verification step from Polarian
```
Comment (quchangl-github) [inline observability/observability_alerts.adoc]: @dockerymick I had updated the Polarion case, when the secret is added into the MCOCR, then user can add below info into the alertmanager.yaml which from secret alertmanager-config
```
"global":
  "http_config":
    "tls_config":
      "cert_file": "/etc/alertmanager/secrets/storyverify/tls.crt"
      "key_file": "/etc/alertmanager/secrets/storyverify/tls.key"
```
And next, save the alertmanager.yaml configuration into the secret alertmanager-config by command:
oc -n open-cluster-management-observability create secret generic alertmanager-config --from-file=alertmanager.yaml --dry-run -o=yaml |  oc -n open-cluster-management-observability replace secret --filename=-

File: observability/observability_alerts.adoc [hunk 4]
```diff
@@ -55,6 +60,86 @@ route:
...
++
+[source,bash]
+----
+openssl x509 -req -days 365 -in cert.csr -signkey cert.key -out cert.crt
+----
+
+//I added the steps to create the .crt and .key files because I saw that in Polarian. Should cert.csr be changed to cert.crt in the openssl req command?
+//
+
+. Mount the `tls` secret to your `MultiClusterObservability` resource by adding it to the `advancedConfig` section. Your resource might resemble the following content:
```
Comment (dockerymick) [inline observability/observability_alerts.adoc]:
```suggestion
. Mount the `tls` secret to your `MultiClusterObservability` resource by adding it to the `advanced` section. Your resource might resemble the following content:
```

File: observability/observability_alerts.adoc [hunk 5]
```diff
@@ -55,6 +60,86 @@ route:
...
+
++
+[source,yaml]
+----
+...
+advancedConfig:
```
Comment (dockerymick) [inline observability/observability_alerts.adoc]:
```suggestion
advanced: 
```

File: observability/observability_alerts.adoc (hunk 2 above)
Comment (dockerymick) [inline observability/observability_alerts.adoc]:
```suggestion
```

File: observability/observability_alerts.adoc [hunk 6]
```diff
@@ -55,6 +60,85 @@ route:
...
+----
+"global":
+  "resolve_timeout": "5m"
+  "http_config":
+    "tls_config":
+      "cert_file": "/etc/alertmanager/secrets/storyverify/tls.crt"
+      "key_file": "/etc/alertmanager/secrets/storyverify/tls.key"
+----
+
+//also got verification step from Polarian
```
Comment (dockerymick) [inline observability/observability_alerts.adoc]: @quchangl-github thanks! It seems like there are commands that you provided. I'll commit my suggestion. Let me know what you think
```suggestion
+
[source,yaml]
----
"global":
  "http_config":
    "tls_config":
      "cert_file": "/etc/alertmanager/secrets/storyverify/tls.crt"
      "key_file": "/etc/alertmanager/secrets/storyverify/tls.key"
----

. Save the `alertmanager.yaml` configuration in the `alertmanager-config` secret by running the following command:

+
[source,bash]
----
oc -n open-cluster-management-observability create secret generic alertmanager-config --from-file=alertmanager.yaml --dry-run -o=yaml 
----

. Replace the previous secret with your new secret by running the following command:

+
[source,bash]
----
oc -n open-cluster-management-observability replace secret --filename=-
----
```

File: observability/observability_alerts.adoc (hunk 1 above)
Comment (dockerymick) [reply observability/observability_alerts.adoc]: @philipgough are the steps to create the `cert.crt` and `cert.key` files necessary?

File: observability/observability_alerts.adoc [hunk 7]
```diff
@@ -3,6 +3,11 @@
 
 Receive and define alerts for the observability service to be notified of hub cluster and managed cluster changes.
 
+[#alert-prereqs]
+== Prerequisites
+
+//are there any prerequsites that we should mention? What is the required access level? Cluster administrator?
```
Comment (dockerymick) [inline observability/observability_alerts.adoc]: @philipgough

File: observability/observability_alerts.adoc [hunk 8]
```diff
@@ -55,6 +60,98 @@ route:
...
+
+. To set an expiration of 365 days for your `cert.crt` and `cert.key` files, run the following command:
+
++
+[source,bash]
+----
+openssl x509 -req -days 365 -in cert.csr -signkey cert.key -out cert.crt
+----
+
+//I added the steps to create the .crt and .key files because I saw that in Polarian. Should cert.csr be changed to cert.crt in the openssl req command?
```
Comment (dockerymick) [inline observability/observability_alerts.adoc]: Necessary to use before the user creates the secret
```suggestion
```

File: observability/observability_alerts.adoc [hunk 9]
```diff
@@ -55,6 +60,98 @@ route:
 
 Your changes are applied immediately after it is modified. For an example of Alertmanager, see link:https://github.com/prometheus/alertmanager/blob/master/doc/examples/simple.yml[prometheus/alertmanager].
 
+[#mount-secrets-alertmanager]
+=== Mounting secrets within the Alertmanager pods
+
+You can create `Secret` resources with arbitrary content, which can be mounted within your `alertmanager` pods for access to authorization credentials.
+
+To reference secrets within your Alertmanager configuration, add the `Secret` resource content within the `open-cluster-management-observability` namespace and mount the content within the `alertmanager` pods. Complete the following steps to create and mount a `tls` secret:
```
Comment (dockerymick) [inline observability/observability_alerts.adoc]:
```suggestion
To reference secrets within your Alertmanager configuration, add the `Secret` resource content within the `open-cluster-management-observability` namespace and mount the content within the `alertmanager` pods. For example, to create and mount a `tls` secret, complete the following steps:
```

File: observability/observability_alerts.adoc [hunk 10]
```diff
@@ -55,6 +60,98 @@ route:
...
+
+. To create create a `tls` secret with TLS certificates, run the following command:
```
Comment (dockerymick) [inline observability/observability_alerts.adoc]:
```suggestion
. To create a `tls` secret with TLS certificates, run the following command:
```

Comment (dockerymick) [inline observability/observability_alerts.adoc]:
```suggestion
. To create `cert.crt` and `cert.key` files, run the following command:
	
+
[source,bash]
----
openssl req -new -newkey rsa:2048 -nodes -keyout cert.key -out cert.csr
----
	
. To set an expiration of 365 days for your `cert.crt` and `cert.key` files, run the following command:
	
+
[source,bash]
----
openssl x509 -req -days 365 -in cert.csr -signkey cert.key -out cert.crt
----

. To create create a `tls` secret with TLS certificates, run the following command:
```

File: release_notes/whats_new.adoc [hunk 11]
```diff
@@ -55,6 +55,8 @@ View tasks and support information at link:../clusters/cluster_mce_overview.adoc
 
 * For more environment stability with default settings, the default CPU request is increased to `500m` and memory request is increased to `1024Mi` for the `thanos-compact` pod. See link:../observability/obs_config.adoc#observability-pod-capacity-requests[Observability pod capacity requests] for more details.
 
+* To create and mount secrets to your `alertmanager` pods for access to authorization credentials, you can add the contents to your `MultiClusterObservability` resource. See link:../observability/observability_alerts.adoc#mount-secrets-alertmanager[Mounting secrets within the Alertmanager pods].
```
Comment (dockerymick) [inline release_notes/whats_new.adoc]:
```suggestion
* To create and mount secrets to your `alertmanager` pods for access to arbitrary content, you can add the contents to your `MultiClusterObservability` resource. See link:../observability/observability_alerts.adoc#mount-secrets-alertmanager[Mounting secrets within the Alertmanager pods].
```

File: observability/observability_alerts.adoc (hunk 7 above)
Comment (dockerymick) [inline observability/observability_alerts.adoc]:
```suggestion
* You must enable observability on your hub cluster.
* You must have the create permission for secret resources in the `open-cluster-management-observability` namespace.
* You must have edit permission on the `MultiClusterObservability` resource.
```

File: observability/observability_alerts.adoc [hunk 12]
```diff
@@ -55,6 +60,98 @@ route:
...
+. To set an expiration of 365 days for your `cert.crt` and `cert.key` files, run the following command:
+
++
+[source,bash]
+----
+openssl x509 -req -days 365 -in cert.csr -signkey cert.key -out cert.crt
+----
+
+//I added the steps to create the .crt and .key files because I saw that in Polarian. Should cert.csr be changed to cert.crt in the openssl req command?
+//
```
Comment (dockerymick) [inline observability/observability_alerts.adoc]:
```suggestion
```

**Review by philipgough — APPROVED** [review]
/lgtm

File: observability/observability_alerts.adoc [hunk 13]
```diff
@@ -55,6 +62,80 @@ route:
...
+. To create a `tls` secret with TLS certificates, run the following command:
+
++
+[source,bash]
+----
+oc create secret tls tls --cert=</path/to/cert.crt> --key=</path/to/cert.key> -n open-cluster-management-observability
+----
+
+
+. Mount the `tls` secret to your `MultiClusterObservability` resource by adding it to the `advanced` section. Your resource might resemble the following content:
```
Comment (jc-berger) [inline observability/observability_alerts.adoc]: Optional:

All your other steps are written in the structure "To < action >, run/add < command or resource > 

Although this command is correct and reads well, consider making its structure consistent with the other steps:

```suggestion
. To mount the `tls` secret to your `MultiClusterObservability` resource, add it to the `advanced` section. Your resource might resemble the following content:
```

File: observability/observability_alerts.adoc [hunk 14]
```diff
@@ -55,6 +62,80 @@ route:
...
+[source,yaml]
+----
+"global":
+  "http_config":
+    "tls_config":
+      "cert_file": "/etc/alertmanager/secrets/storyverify/tls.crt"
+      "key_file": "/etc/alertmanager/secrets/storyverify/tls.key"
+----
+
+. Save the `alertmanager.yaml` configuration in the `alertmanager-config` secret by running the following command:
```
Comment (jc-berger) [inline observability/observability_alerts.adoc]: Same optional comment as above. Simple sentence restructure for consistency. Once again, original content is good and correct :) 

```suggestion
. To save the `alertmanager.yaml` configuration in the `alertmanager-config` secret, run the following command:
```

File: observability/observability_alerts.adoc [hunk 15]
```diff
@@ -55,6 +62,80 @@ route:
...
+
++
+[source,bash]
+----
+oc -n open-cluster-management-observability create secret generic alertmanager-config --from-file=alertmanager.yaml --dry-run -o=yaml 
+----
+
+. Replace the previous secret with your new secret by running the following command:
```
Comment (jc-berger) [inline observability/observability_alerts.adoc]:
```suggestion
. To replace the previous secret with your new secret, run the following command:
```

**Review by jc-berger — APPROVED** [review]
/lgtm

**Pull request merged**